src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/UpdateApprenticeshipStopDateCommandHandlerTests.cs

```diff
@@ -0,0 +1,387 @@
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Domain.Entities;
+
+namespace SFA.DAS.CommitmentsV2.UnitTests.Application.Commands
+{
+    [TestFixture]
+    [Parallelizable]
+    public class UpdateApprenticeshipStopDateCommandHandlerTests
+    {
+        private Mock<ICurrentDateTime> _currentDateTime;
+        private Mock<ILogger<UpdateApprenticeshipStopDateCommandHandler>> _logger;
+        private Mock<IAuthenticationService> _authenticationService;
+        private Mock<IMessageSession> _nserviceBusContext;
+        private Mock<IEncodingService> _encodingService;
+        private Mock<IOverlapCheckService> _overlapCheckService;
+        ProviderCommitmentsDbContext _dbContext;
+        ProviderCommitmentsDbContext _confirmationDbContext;
+        private UnitOfWorkContext _unitOfWorkContext { get; set; }
+        private IRequestHandler<UpdateApprenticeshipStopDateCommand> _handler;
+
+        [SetUp]
+        public void Init()
+        {
+            var databaseGuid = Guid.NewGuid().ToString();
+            _dbContext = new ProviderCommitmentsDbContext(new DbContextOptionsBuilder<ProviderCommitmentsDbContext>()
+                                        .UseInMemoryDatabase(databaseGuid)
+                                        .ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.QueryClientEvaluationWarning))
+                                        .Options);
+
+            _confirmationDbContext = new ProviderCommitmentsDbContext(new DbContextOptionsBuilder<ProviderCommitmentsDbContext>()
+                            .UseInMemoryDatabase(databaseGuid)
+                            .ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.QueryClientEvaluationWarning))
+                            .Options);
+
+            _currentDateTime = new Mock<ICurrentDateTime>();
+            _authenticationService = new Mock<IAuthenticationService>();
+            _nserviceBusContext = new Mock<IMessageSession>();
+            _encodingService = new Mock<IEncodingService>();
+            _overlapCheckService = new Mock<IOverlapCheckService>();
+            _overlapCheckService.Setup(x => x.CheckForOverlaps(It.IsAny<string>(), It.IsAny<CommitmentsV2.Domain.Entities.DateRange>(), It.IsAny<long?>(), It.IsAny<CancellationToken>()));
+            _logger = new Mock<ILogger<UpdateApprenticeshipStopDateCommandHandler>>();
+            _unitOfWorkContext = new UnitOfWorkContext();
+
+            _handler = new UpdateApprenticeshipStopDateCommandHandler(new Lazy<ProviderCommitmentsDbContext>(() => _dbContext),
+                _logger.Object,
+                _currentDateTime.Object,
+                _authenticationService.Object,
+                 _nserviceBusContext.Object,
+                _encodingService.Object,
+                _overlapCheckService.Object);
+                
+        }
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_WithInvalidData_ThenShouldThrowException()
```

Comment:
Name of this test suggests it should throw an exception?

Reply:
changed the method name